driver-core/src/main/com/mongodb/session/ClientSession.java

```diff
@@ -56,6 +56,35 @@
      */
     void setPinnedServerAddress(ServerAddress address);
 
+    /**
+     * Gets the transaction context.
+     *
+     * <p>For internal use only </p>
+     *
+     * @return the transaction context
+     */
+    @Nullable
+    Object getTransactionContext();
```

Comment:
Slightly ugly using Object, did you consider adding a TransactionContext interface?

Reply:
I did consider this.  My reasoning is that, since it's an internal API, there is no need to introduce a new type for the parameter, since this new type itself should not be public (nor did I want to make ReferenceCounted a public type).